sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs

```diff
@@ -285,13 +285,14 @@ public async Task PartitionPumpCreatesScopeForEventProcessing()
             var clientMock = new Mock<EventHubClient>();
             clientMock.Setup(c => c.CreateConsumer("cg", "pid", It.IsAny<EventPosition>(), It.IsAny<EventHubConsumerOptions>())).Returns(consumerMock.Object);
 
-            var processorMock = new Mock<BasePartitionProcessor>();
-            processorMock.Setup(p => p.InitializeAsync(It.IsAny<PartitionContext>())).Returns(Task.CompletedTask);
-            processorMock.Setup(p => p.ProcessEventsAsync(It.IsAny<PartitionContext>(), It.IsAny<IEnumerable<EventData>>(), It.IsAny<CancellationToken>()))
-                .Returns(Task.CompletedTask)
-                .Callback(() => processorCalledSource.SetResult(null));
+            var eventProcessorMock = new Mock<EventProcessor>();
+            eventProcessorMock.Object.ProcessEventsAsync = (context, manager) =>
+            {
+                processorCalledSource.SetResult(null);
+                return Task.CompletedTask;
+            };
 
-            var manager = new PartitionPump(clientMock.Object, "cg", new PartitionContext("ns", "eh", "cg", "pid", "oid", new InMemoryPartitionManager()), processorMock.Object, new EventProcessorOptions());
+            var manager = new PartitionPump(eventProcessorMock.Object, clientMock.Object, "cg", new PartitionContext("ns", "eh", "cg", "pid", "oid", new InMemoryPartitionManager()), new EventProcessorOptions());
```

Comment:
`InMemoryPartitionManager` is marked for death, yeah?

Reply:
I don't know. Is it? If we kill it, we can just add a mock here.

Reply:
Taking a look at the new API, it makes sense to kill it, but that's something we'll end up doing only after preview 5.